### ISSUE TITLE
Wanted to get rid of a warning

### ISSUE THREAD

**Issue by flip111** [ku-fpg/dotgen#3]
I saw this warning

```
dotgen                     > Preprocessing library for dotgen-0.4.2..
dotgen                     > Building library for dotgen-0.4.2..
dotgen                     > [1 of 1] Compiling Text.Dot
dotgen                     >                
dotgen                     > /tmp/stack22672/dotgen-0.4.2/Text/Dot.hs:157:9: warning: [-Wincomplete-patterns]
dotgen                     >     Pattern match(es) are non-exhaustive
dotgen                     >     In an equation for ‘showAttrs'’: Patterns not matched: []
dotgen                     >     |          
dotgen                     > 157 |         showAttrs' [a]    = showAttr a
dotgen                     >     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```

I wanted to send a pull request to do something about it, but i didn't know the right way to go about it. Some options:

* Disable warnings for unused patterns for the entire module (easy but dangerous)
* Pattern match on `[]` and throw an error (not sure if there are any downsides to this)
* Change the design so `NonEmptyList` is used

I can look into sending a PR after some advice perhaps.

**Comment by RyanGlScott** [comment]
Thanks for looking into this! Given that `showAttrs` already upholds the invariant that `showAttrs'` argument is non-empty, I think the second option (Pattern match on [] and throw an error) would suffice. The third option (Change the design so `NonEmptyList` is used) would also be fine.

**Comment by RyanGlScott** [comment]
Fixed in #4.
